src/daria/image/image.py

```diff
@@ -49,38 +50,68 @@ class Image:
-            read_metadata_from_file (bool): controlling whether meta data is read from file
-            metadata_path (str): path to metadata; should be changed if metadata is not placed
-                in the "images/metadata"-folder and taking the same path as the image. If
-                image is not passed as a path one also needs to pass it as a path.
+            img (Union[np.ndarray, str]): image array with matrix indexing
+            metadata (dict, Optional): metadata dictionary, default is None.
+            curvature_correction (daria.CurvatureCorrection, Optional):
+                Curvature correction object. Default is none, but should be included
+                if the image is to be curvature corrected at initialization
+            color_correction (daria.ColorCorrection, Optional): Color correction object.
+                Default is none, but should be included if the image is to be color
+                corrected at initialization.
+            kwargs (Optional arguments)
+                metadata_source (str): Path to a metadata json-file that provides
+                    metadata such as physical width, height and origo of image
+                    as well as colorspace
+                origo (np.ndarray): physical coordinates of the lower left corner
+                width (float): physical width of the image
+                height (float): physical height of the image
+                color_space (str): the colorspace of the image. So far only BGR
+                    and RGB are "valid", but more should be added at a later time.
         """
 
+        # Read metadata
+        no_colorspace_given = False
+        if metadata is not None:
+            self.metadata = metadata
+            self.width: float = self.metadata["width"]
+            self.height: float = self.metadata["height"]
+            self.origo: list[float] = self.metadata["origo"]
+            self.colorspace: str = self.metadata["color_space"]
+
+        elif "metadata_source" in kwargs:
+            metadata_source = kwargs.pop("metadata_source")
+            with open(str(Path(metadata_source)), "r") as openfile:
+                self.metadata = json.load(openfile)
+
+            self.width: float = self.metadata["width"]
+            self.height: float = self.metadata["height"]
+            self.origo = self.metadata["origo"]
+            self.colorspace: str = self.metadata["color_space"]
+
+        else:
+            self.width: float = kwargs.pop("width", 1)
+            self.height: float = kwargs.pop("height", 1)
+            self.origo: list[float] = kwargs.pop("origo", [0, 0])
+            if "color_space" in kwargs:
+                self.colorspace = kwargs["color_space"]
+            else:
+                self.colorspace = "BGR"
+                no_colorspace_given = True
+            self.colorspace: str = kwargs.pop("color_space", "BGR")
```

Comment:
Here, one would have to feed the keyword arguments to `update_metadata`.

Reply:
Or simply directly create `self.metadata = {}` and then line by line fill in `width`, etc.